db/db_bolt.go

```diff
@@ -86,6 +88,9 @@ func (b *BoltDB) Put(namespace string, key, value []byte) (err error) {
 		}
 	}
 	if err != nil {
+		if errors.Is(err, syscall.ENOSPC) {
```

Comment:
or should we use `errors.Cause(err) == syscall.ENOSPC`? can you look at bolt db source code to verify?

Reply:
`errors.Cause(err)` returns `write /run/data/chain.db: no space left on device`
`syscall.ENOSPC` is `no space left on device`
`errors.Is` means `syscall.ENOSPC` matches `err`

Reply:
errors.Is(erros.Cause(), err)

Reply:
```
// An error is considered to match a target if it is equal to that target or if
// it implements a method Is(error) bool such that Is(target) returns true.
func Is(err, target error) bool { return stderrors.Is(err, target) }
```
i think better to use `errors.Cause() == xxx` if possible? this is how error is decided in the codebase, try to be consistent

Reply:
```
// An error is considered to match a target if it is equal to that target or if
// it implements a method Is(error) bool such that Is(target) returns true.
func Is(err, target error) bool { return stderrors.Is(err, target) }
```

can you verify (or write a test case) this is the case?